com.unity.render-pipelines.universal/Runtime/Passes/MainLightShadowCasterPass.cs

```diff
@@ -230,7 +230,10 @@ void SetupMainLightShadowReceiverConstants(CommandBuffer cmd, VisibleLight shado
                     m_CascadeSplitDistances[3].w * m_CascadeSplitDistances[3].w));
             }
 
-            if (softShadows)
+            // Inside shader soft shadows are controlled through global keyword.
+            // If any additional light has soft shadows it will force soft shadows on main light too.
+            // TODO: make this conditional once we will support soft shadows per light inside shader.
+            // if (softShadows)
```

Comment:
This will cause these constants to be uploaded even when no light with soft shadows is visible. What if we change the condition here to check only for `supportsSoftShadows` for now?
https://github.com/Unity-Technologies/Graphics/blob/master/com.unity.render-pipelines.universal/Runtime/Passes/MainLightShadowCasterPass.cs#L179

Reply:
Yea that makes more sense, added check in latest commit.